libwacom/libwacom-database.c

```diff
@@ -336,6 +349,34 @@ libwacom_parse_stylus_keyfile(WacomDeviceDatabase *db, const char *path)
 		g_key_file_free (keyfile);
 }
 
+static void
+libwacom_setup_paired_attributes(WacomDeviceDatabase *db)
+{
+	GHashTableIter iter;
+	gpointer key, value;
+
+	g_hash_table_iter_init(&iter, db->stylus_ht);
+	while (g_hash_table_iter_next (&iter, &key, &value)) {
+		WacomStylus *stylus = value;
+		const int* ids = libwacom_stylus_get_paired_ids(stylus);
+		int count = libwacom_stylus_get_num_paired_ids(stylus);
+		int i;
+
+		for (i = 0; i < count; i++) {
+			const WacomStylus *pair;
+
+			pair = libwacom_stylus_get_for_id(db, ids[i]);
+			if (pair == NULL) {
+				g_warning("Paired stylus '0x%x' not found, ignoring.", ids[i]);
```

Comment:
fwiw, printf directive `%#x` prints a hex number prefixed with `0x`

Reply:
:+1:

Reply:
i noticed you fixed this in the hunk above but not here - any specific reason for that?